[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61782

This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

Commit:
[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests

Summary:
Added more comprehensive tests for Android NNAPI delegate.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


Summary:
Added more comprehensive tests for Android NNAPI delegate.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


Summary:
Added more comprehensive tests for Android NNAPI delegate.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

dreiss commented on test/test_nnapi.py:
```diff
@@ -32,6 +32,14 @@ def setUp(self):
+        return convert_model_to_nnapi(traced_module, args)
+
+    # Created for subclasses to set can_run_nnapi (eg TestNnapiBackend)
+    def set_can_run_nnapi(self, bool):
```
Don't call this argument bool, since that shadows a built-in name.  Maybe `can_run`.

dreiss commented on test/jit/test_backend_nnapi.py:
```diff
@@ -29,73 +29,49 @@
+without the delegate API.
 """
 # This is needed for IS_WINDOWS or IS_MACOS to skip the tests.
 @unittest.skipIf(TEST_WITH_ROCM or IS_SANDCASTLE or IS_WINDOWS or IS_MACOS or IS_FBCODE,
```
Can we just skip the execution but still test the conversion process, so we don't need all of these skip directives?

Amyh11325 replied:
I think the execution is already skipeed. I thought we'd still need to test the execution later, unless you want me to make separate tests for testing execution and testing conversion?

iseeyuan replied:
@dreiss I think you've already noticed in your other comment, that the skip is not to skip execution, but the entire end-to-end process. The reason is that both preprocess and execute functions are built into a .so file, but not directly into pytorch. We could change it later when we decide to build the delegation lib to pytorch.

dreiss commented on test/jit/test_backend_nnapi.py:
```diff
@@ -29,73 +29,49 @@
-        traced = torch.jit.trace(module, args)
-        compile_spec = {"forward": {"inputs": args}}
-        nnapi_model = torch._C._jit_to_backend("nnapi", traced, compile_spec)
+        # Load nnapi delegate library
+        torch_root = Path(__file__).resolve().parent.parent.parent
+        p = torch_root / 'build' / 'lib' / 'libnnapi_backend.so'
+        torch.ops.load_library(str(p))
```
Can we just build this into the normal PyTorch build so we don't need a separate load step?  This doesn't seem portable.

Amyh11325 replied:
I don't quite know how to build this into the normal PyTorch build or what it means. I copied this step from what's in test_backends.py. @iseeyuan @raziel Any thoughts?

iseeyuan replied:
I thought building an .so file and loading it from python is a format used for the backends, to show the flexibility of loading different backends as needed. But you are right, it's not portable for testing purpose. @Amyh11325 could you add an action item in your doc, so we could follow up when you finish the milestones?

raziel replied:
So, it's the plan to keep testing with the .so for the base infra tests, and then the specific backends tests to do it in the PyTorch build?

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"

This PR depends on https://github.com/pytorch/pytorch/pull/61787

### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).

iseeyuan commented on test/jit/test_backend_nnapi.py:
```diff
@@ -29,73 +28,49 @@
-        super().__init__(name)
-        self.prelu_test = NnapiBackendPReLUTest(name)
+    # Override
+    def call_convert_to_nnapi(self, traced_module, args):
```
Could you change this function name to "test_delegate_to_nnapi", to differentiate it from the original convert method?

Amyh11325 replied:
I'm using this method to override one in test_nnapi.py, so I don't think that name works because in test_nnapi.py the method doesn't use the delegate. Maybe "test_lowering_to_nnapi" is better?

Amyh11325 replied:
Oh, the name can't start with test or else the TestCase will try to run the function without arguments. How about "call_lowering_to_nnapi"?

iseeyuan reviewed (approved):
Please create a task on following up the issue of skipping platforms and put it in your doc.

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

raziel reviewed (approved):
Thanks.
I think David's comments make sense, but we should address them in a specific PR (and decide whether this makes sense also for test_backends.py --maybe not).

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).

Amyh11325 commented:
Merging with master now that https://github.com/pytorch/pytorch/pull/61878 has landed.

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).

Amyh11325 commented:
Fixing CI error, by stopping macOS from importing test_nnapi.py.

axitkhurana commented on test/jit/test_backend_nnapi.py:
```diff
@@ -1,18 +1,16 @@
-    TEST_WITH_ASAN
-)
+# Cannot import test_nnapi on Apple devices
+if (sys.platform != "Darwin"):
```
Don't need () in python

Amyh11325 replied:
I never noticed, wow...

axitkhurana commented on test/jit/test_backend_nnapi.py:
```diff
@@ -26,75 +24,53 @@
+# First skip is needed for IS_WINDOWS or IS_MACOS to skip the tests.
+# Second skip is because ASAN is currently causing an error.
+# It is still unclear how to resolve this. T95764916
 @unittest.skipIf(not os.path.exists(Path(__file__).resolve().parent.parent.parent
```
should we get the path in a separate variable we can inside the test too?

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).

iseeyuan commented on aten/src/ATen/nnapi/nnapi_bind.cpp:
```diff
@@ -214,6 +223,8 @@ static auto register_NnapiCompilation = [](){
     throw;
   }
 }();
+#else
+  #undef IS_IOS_NNAPI_BIND
```
These two lines are not necessary. If it's not defined, you don't have to undefine it.

Amyh11325 replied:
I think they are necessary, because if it's not-not defined, then undefine it.

iseeyuan replied:
Ah, make sense! I read it wrong.

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Commit:
Update on "[6/N] Nnapi Backend Delegate: Comprehensive OSS Tests"


This PR depends on https://github.com/pytorch/pytorch/pull/61787
This PR adds OSS tests for https://github.com/pytorch/pytorch/pull/61499, which was previously only tested locally
### Summary:
Added more comprehensive tests for Android NNAPI delegate.
Previously, there was only one basic test for lowering a PReLU module with the NNAPI delegate. Now, more tests are inherited from `test_nnapi.py`, the file for testing NNAPI conversion and execution without the delegate.

**test_backend_nnapi.py**
Test file for Android NNAPI delegate. 
- `TestNnapiBackend` class inherits tests from `test_nnapi.py` and overrides the model conversion to use the delegate API.
- Includes an extra test for passing input arguments as Tensors and Tensor Lists. 
- Has extra set up for loading the NNAPI delegate library and changing the default dtype from float64 to float32 (dtype is typically float32 by default, but not in delegate backend unit tests)

**test_nnapi.py**
Test file for Android NNAPI without the delegate. 
- Some code was refactored to allow override of only the NNAPI conversion call. 
- An extra function was added to allow the NNAPI delegate unit test to turn off the model execution step. Once the NNAPI delegate's execution implementation is complete, this may no longer be necessary.

### Test Plan:
I ran `python test/test_jit.py TestNnapiBackend` and `python test/test_nnapi.py` to run both test files.

Reviewers:

Subscribers:

Tasks:
T91991864
Tags:

Differential Revision: [D29772005](https://our.internmc.facebook.com/intern/diff/D29772005)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772005).
